easybuild/tools/module_generator.py

```diff
@@ -185,7 +186,8 @@ def prepare(self, fake=False):
         mkdir(os.path.dirname(mod_filepath), parents=True)
 
         # remove module file if it's there (it'll be recreated), see EasyBlock.make_module
-        if os.path.exists(mod_filepath) and not build_option('extended_dry_run'):
+        if os.path.exists(mod_filepath) and not build_option('extended_dry_run') and \
+                (not build_option('backup_modules') or fake):
             self.log.debug("Removing existing module file %s", mod_filepath)
             os.remove(mod_filepath)
```

Comment:
Hmm, isn't this duplicate code since we're already removing the module file in `check_readiness_step` as well?

Also here: we should rename here already rather than not remove?